core/modules/rproc/ProtoRowBuffer.h

```diff
@@ -23,14 +23,118 @@
 #ifndef LSST_QSERV_RPROC_PROTOROWBUFFER_H
 #define LSST_QSERV_RPROC_PROTOROWBUFFER_H
 
+// System headers
+#include <limits>
+
+
+// Qserv headers
 #include "mysql/RowBuffer.h"
 #include "proto/worker.pb.h"
+#include "sql/Schema.h"
 
 namespace lsst {
 namespace qserv {
 namespace rproc {
-/// Construct a RowBuffer with a Result message row source.
-mysql::RowBuffer::Ptr newProtoRowBuffer(proto::Result& r);
+
+
+/// ProtoRowBuffer is an implementation of RowBuffer designed to allow a
+/// LocalInfile object to use a Protobufs Result message as a row source
+class ProtoRowBuffer : public mysql::RowBuffer {
+public:
+    ProtoRowBuffer(proto::Result& res, int jobId, std::string const& jobIdColName,
+                   std::string const& jobIdSqlType, int jobIdMysqlType);
+    virtual unsigned fetch(char* buffer, unsigned bufLen);
+    std::string dump() const override;
+
+    /// Escape a bytestring for LOAD DATA INFILE, as specified by MySQL doc:
+    /// https://dev.mysql.com/doc/refman/5.1/en/load-data.html
+    /// This is limited to:
+    /// Character    Escape Sequence
+    /// \0     An ASCII NUL (0x00) character
+    /// \b     A backspace character
+    /// \n     A newline (linefeed) character
+    /// \r     A carriage return character
+    /// \t     A tab character.
+    /// \Z     ASCII 26 (Control+Z)
+    /// \N     NULL
+    ///
+    /// @return the number of bytes written to dest
+    template <typename Iter, typename CIter>
+    static inline int escapeString(Iter destBegin, CIter srcBegin, CIter srcEnd) {
```

Comment:
I don't know. I moved the definition of this class into the header file from the cc file and I did not want to alter what I believe to be working code any more than really necessary.